app/models/host_report.rb

```diff
@@ -55,4 +55,53 @@ def self.search_by_keyword(_key, operator, value)
       conditions: sanitize_sql_for_conditions(["host_reports.report_keyword_ids @> ?", "{#{keyword_ids.join(',')}}"]),
     }
   end
+
+  def self.report
+    @report = {} if @report.nil?
+    @report.update(
+      :total_hosts => Host.all.count,
+      :change_hosts => report_changes.count,
+      :nochange_hosts => report_nochanges.count,
+      :failure_hosts => report_failures.count,
+      :disabled_hosts => disabled_alerts.count,
+      :reports_missing => report_missing.count
+    )
+  end
```

Comment:
Probably try to find a better method name - this is HostReport model and it is weird. Perhaps `totals` or `counts` or something like that. The helper methods should be `private`.

Reply:
My comment still stands.